scripts/Crawlers/BaseCrawler.py

```diff
@@ -552,3 +558,29 @@ def _check_dats_present(self, directory):
                         return os.path.join(file_path, subfile_name)
             elif file_name.lower() == "dats.json":
                 return file_path
+
+    def _add_source_data_submodule_if_derived_from_conp_dataset(
+        self, dats_json, dataset_dir
+    ):
+        with open(dats_json) as f:
+            metadata = json.loads(f.read())
+
+        source_dataset_link = None
+        source_dataset_id = None
+        if "extraProperties" not in metadata.keys():
+            return
+        for property in metadata["extraProperties"]:
+            if "values" not in property.keys():
+                continue
+            if type(property["values"]) != list:
+                continue
+            if "value" not in property["values"][0].keys():
+                continue
+            if property["category"] == "derivedFrom":
+                source_dataset_link = property["values"][0]["value"]
+            if property["category"] == "parent_dataset_id":
+                source_dataset_id = property["values"][0]["value"]
+
+        if source_dataset_link and "github.com" in source_dataset_link:
```

Comment:
I would be explicit here and check that `source_dataset_link is not None`. `None` is "falsy" and here the variable is declared just above, but it's still better to [be consistent](https://legacy.python.org/dev/peps/pep-0008/#programming-recommendations).

Reply:
👍  Done in the last commit